### ISSUE TITLE
run capslock against a module's url

### ISSUE THREAD

**Issue by plastikfan** [google/capslock#200]
If the point of this program is to protect against supply chain attacks and to help developers know if it's safe to use a particluar dependency, why do we have to download the target dependency first? Would it not be better to be able to run capslock against a url, eg:

> caplock github.com/scammer/dodgy-package

and then get a report displayed locally?

**Comment by jcd2** [comment]
I agree, it would be better if users didn't need to go through extra steps and add the dependency to one of their modules, just so they can run capslock on it.

Capslock does need the package to be added as a dependency to a module _somewhere_, so that it can point the Go tools at that module and ask it to determine the right files and transitive dependencies for Capslock to analyze.  So I've made a change where Capslock will automatically take care of creating a module with the right dependencies in a temporary directory, so that it can analyze them and then delete the directory afterwards.  Then we'll be able to run Capslock on any package and it will handle it.
